### ISSUE TITLE
Invalid html generated

### ISSUE THREAD

**Issue by tdelmas** [gohugoio/hugo#9159]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.89.2-63E3A5EB+extended linux/amd64 BuildDate=2021-11-08T15:22:24Z VendorInfo=gohugoio 
</pre>

### Does this issue reproduce with the latest release?

Yes.

That issue was not present in Hugo 0.88.1 and is present at least since 0.89.1

###

Minimal template to reproduce the issue:

```markdown
---
slug: bug
---

test <a
href="https://exemple.com/">link</a> end.

```

**Important**: new lines of that example are CRLF (\r\n), **NOT** \n. There is no issue when the newlines are \n.

## Html generated

```html
<p>test &lt;a
href=&ldquo;<a href="https://exemple.com/%22%3elink">https://exemple.com/">link</a></a> end.</p>
```

## Html expected (and generated by Hugo 0.88.1)

```html
<p>test <a
href="https://exemple.com/">link</a> end.</p>
```

I tried to reproduce that issue with The goldmark version of Hugo  0.89.2 but without success.

**Comment by bep** [comment]
>I tried to reproduce that issue with The goldmark version of Hugo 0.89.2 but without success.

Are you saying it's fixed in 0.89.2?

**Comment by tdelmas** [comment]
> Are you saying it's fixed in 0.89.2?

No, The issue is present in Hugo 0.89.1 **and* 0.89.2.

But, I didn't succeed to reproduce it in goldmark. I don't know if:
- I just made a mistake when testing with goldmark (I'm not fluent in go)
- Or the bug is not in goldmark but in the way Hugo transmits the file content to goldmark

**Comment by jmooring** [comment]
https://github.com/yuin/goldmark/issues/263

**Comment by tdelmas** [comment]
I think it should be fixed with Goldmark 1.4.4
